packages/framework/src/Component/Image/Processing/ImageProcessingService.php

```diff
@@ -79,9 +96,13 @@ public function convertToShopFormatAndGetNewFilename($filepath)
             throw new \Shopsys\FrameworkBundle\Component\Image\Processing\Exception\FileIsNotSupportedImageException($filepath);
         }
 
-        $image = $this->createInterventionImage($filepath)->save($newFilepath);
+        $image = $this->createInterventionImage($filepath);
+
+        $image->encode();
+        $this->filesystem->put($newFilepath, $image);
+
         if (realpath($filepath) !== realpath($newFilepath)) {
-            $this->filesystem->remove($filepath);
+            $this->localFilesystem->remove($filepath);
```

Comment:
This is a strange, the file can be in filesystem, or local filesystem, but we remove it only from local filesystem.
I think we should know where it was and then behave by this information.